storage_v2/factoryadapter/reader.go

```diff
@@ -62,8 +63,30 @@ func (tr *TraceReader) GetOperations(ctx context.Context, query tracestore.Opera
 	return operations, nil
 }
 
-func (*TraceReader) FindTraces(_ context.Context, _ tracestore.TraceQueryParameters) ([]ptrace.Traces, error) {
-	panic("not implemented")
+func (tr *TraceReader) FindTraces(ctx context.Context, query tracestore.TraceQueryParameters) ([]ptrace.Traces, error) {
+	t, err := tr.spanReader.FindTraces(ctx, &spanstore.TraceQueryParameters{
+		ServiceName:   query.ServiceName,
+		OperationName: query.OperationName,
+		Tags:          query.Tags,
+		StartTimeMin:  query.StartTimeMin,
+		StartTimeMax:  query.StartTimeMax,
+		DurationMin:   query.DurationMin,
+		DurationMax:   query.DurationMax,
+		NumTraces:     query.NumTraces,
+	})
+	if err != nil || t == nil {
+		return nil, err
+	}
+	otelTraces := []ptrace.Traces{}
+	for _, trace := range t {
+		batch := &model.Batch{Spans: trace.GetSpans()}
+		otelTrace, err := model2otel.ProtoToTraces([]*model.Batch{batch})
```

Comment:
@yurishkuro is this a good way to perform the conversion?

Reply:
lgtm. What's the concern?

Reply:
`model.Trace` also has `ProcessMap` and `Warnings` which don't get translated over - is that okay?

Reply:
I don't think those are ever populated by storage - rather they are populated by the query service before returning to the UI.